### PR TITLE
[Merged by Bors] - feat: remove completeness assumption in the formula for `∑ n, n * r ^ n`

### PR DESCRIPTION
**Pull request by sgouezel**
Currently, the formula `∑ n, n * r ^ n = r / (1 - r) ^ 2` in a field assumes that the field is complete to get summability of the series from the summability of the norms. However, as `∑ r ^ n` is converging (to `1 / (1 - r)`), the summability of the Cauchy product is automatic thanks to a variation around results we already have, so completeness can be removed. 

In the same way, `∑ n ^ k * r ^ n` is summable without completeness assumptions. The proof is a little bit indirect (first get summability of `∑ n, (n + k).choose k * r ^ n` for each `k`, as this is `(∑ r ^ n) ^ k`, and then deduce the other one), but it gives the nice formula `∑ n, (n + k).choose k * r ^ n = 1 / (1 - r) ^ k` along the way, which is nice to have.

This is a preliminary PR before unifying several results between `[NormedRing R] [CompleteSpace R]` and `[NontriviallyNormedField R]`, which are proved twice with very similar arguments at several places in mathlib.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
Thank you for working on this! It was on my low-priority TODO list for a long time, see #5878

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (Parcly-Taxel) [inline Mathlib/Data/Nat/Choose/Sum.lean:139]: Can you note that this is just a shifted version of my `sum_Icc_choose` right above?

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (Parcly-Taxel) [inline Mathlib/Data/Nat/Choose/Sum.lean:147]:
```suggestion
/-- **Zhu Shijie's identity** aka hockey-stick identity. Version with `range` instead of `Icc`. -/
lemma sum_range_add_choose (n k : ℕ) :
    ∑ i ∈ range (n + 1), (i + k).choose k = (n + k + 1).choose (k + 1) := by
  rw [← sum_Icc_choose (n + k) k, range_eq_Ico]
  convert (sum_map _ (addRightEmbedding k) (·.choose k)).symm using 2
  rw [map_add_right_Ico, zero_add, add_right_comm]; rfl
```

Comment (Parcly-Taxel) [reply Mathlib/Data/Nat/Choose/Sum.lean:147]: In fact I've opened #15935 to add this lemma properly.

Comment (sgouezel) [reply Mathlib/Data/Nat/Choose/Sum.lean:147]: Thanks for your shorter proof! However, you shouldn't have removed the combinatorial interpretation, because I'm pretty sure people looking for this lemma will grep for "stars and bars", so I'm adding it back.

File: Mathlib/Analysis/SpecificLimits/Normed.lean
Comment (Parcly-Taxel) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:394]:
```suggestion
        simp only [this, ← sum_mul, ← Nat.cast_sum, sum_range_add_choose n k]
```

**Comment by sgouezel** [comment]
@Parcly-Taxel,you shouldn't have pushed to the branch. In general, don't ever push to a branch unless it's yours, or if it's tagged `help-wanted`. The change you have done removing the parentheses doesn't look good, because `k + 1` is a single unit, so it was kept together on purpose.

If you have suggestions to make, the correct way is to suggest these using the github interface, and let the owner of the branch accept them or not.

By the way, thanks a lot for the review, this is much appreciated and very useful!

File: Mathlib/Algebra/BigOperators/Fin.lean
Comment (urkud) [inline Mathlib/Algebra/BigOperators/Fin.lean:96]: Is it a good `simp` lemma? Same question about `prod_cons`.

File: Mathlib/Order/Filter/AtTopBot.lean
Comment (urkud) [inline Mathlib/Order/Filter/AtTopBot.lean:1318]: Given the next lemma, should it be `Tendsto _ atTop atTop`?

File: Mathlib/Data/Nat/Factorial/BigOperators.lean
Comment (urkud) [inline Mathlib/Data/Nat/Factorial/BigOperators.lean:42]: Does it make sense to define this as a `Polynomial Nat`?

File: Mathlib/Data/Nat/Choose/Bounds.lean
Comment (urkud) [inline Mathlib/Data/Nat/Choose/Bounds.lean:40]: Please add a docstring saying that this name was used for a lemma which is now called `choose_le_pow_div`.

File: Mathlib/Data/Nat/Factorial/BigOperators.lean
Comment (sgouezel) [reply Mathlib/Data/Nat/Factorial/BigOperators.lean:42]: Indeed, it's ascPochhammer up to a shift, so I have deleted this and used directly ascPochhammer in the main proof. Thanks!

File: Mathlib/Algebra/Polynomial/Degree/Lemmas.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Polynomial/Degree/Lemmas.lean:356]:
```suggestion
        ne_zero_of_natDegree_gt (Nat.pos_of_ne_zero q0), not_false_eq_true, pow_ne_zero, or_self]
```

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Nat/Choose/Sum.lean:142]: What's the `rfl` doing? Is there a missing lemma?

Comment (sgouezel) [reply Mathlib/Data/Nat/Choose/Sum.lean:142]: It's for `Icc k (n + k) = Ico k (n + k + 1)`. We have the `Set` version, in `Order.Ico_succ_right` (not imported by this file, though), but not the `Finset` version, so I'm happy to leave that to `rfl`. The other option is to port the material on set intervals from `Mathlib.Order.SuccPred.Basic`  to finset intervals, but I think it's a little bit too much for the current PR.

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/Nat/Choose/Sum.lean
Comment (urkud) [reply Mathlib/Data/Nat/Choose/Sum.lean:142]: Is it `Nat.Ico_succ_right`?

Comment (urkud) [reply Mathlib/Data/Nat/Choose/Sum.lean:142]: Otherwise LGTM. Thanks a lot!
bors d+

**Comment by sgouezel** [comment]
bors r+
Thanks for the review!

**Pull request closed without merging**
